### DISCUSSION TITLE
Can simnibs import other models, such as a spherical model?

### DISCUSSION THREAD

**Discussion by lingoodluck** [simnibs/simnibs#459] [2. General]
If possible, how to generate the import data?

**Comment by TorgeW** [comment]
Yes there is a spherical model in the example dataset ("m2m_sphere") https://simnibs.github.io/simnibs/build/html/dataset.html
You can also generate your own tetrahedral meshes in msh format. You just have to make sure that the tags are correct (https://github.com/simnibs/simnibs/blob/master/simnibs/utils/mesh_element_properties.py) and that the tetrahedral quality is good.

**Reply by lingoodluck** [reply]
I have used a spherical model and performed TI calculations to generate an msh file. How can I convert it into a three-dimensional npy file? I want to use it for training in a neural network. The npy file exported directly from GMSH shows as corrupted.

**Reply by TorgeW** [reply]
Please look into the tutorials (on the simnibs website) and examples (inside the example folder). 
You will see how you can load and work with data that is present in the mesh. From this you will be able to export it with numpy.

**Reply by lingoodluck** [reply]
Hello, I conducted two TDCS simulations using a spherical model. The two electrodes are symmetrical with respect to the diameter of the sphere, and I set the same reference point, but the electric fields they generate are not symmetrical. I think there is an issue here. Is this normal? Here is my code:

S = sim_struct.SESSION()
S.fnamehead = "D:/simnibs_project/m2m_sphere_twolayer/myspheres_twolayer.msh"
S.pathfem = 'D:/simnibs_project/m2m_sphere_twolayer/TI'  # Directory for the simulation


tdcs = S.add_tdcslist()
tdcs.currents = [0.01, -0.01,0]  # Current flow though each channel (A)

electrode = tdcs.add_electrode()
electrode.channelnr = 1
electrode.centre = [100-math.sqrt(9375),75,100]#
electrode.shape = 'ellipse'
electrode.dimensions = [10,10]  # diameter in [mm]
electrode.thickness = 2#[2,1,2]  # 1 mm thickness
#electrode.dimensions_sponge = [20,20]


electrode = tdcs.add_electrode()
electrode.channelnr = 2
electrode.centre = [100-math.sqrt(9375),125,100]
electrode.shape = 'ellipse'
electrode.dimensions = [10,10]  # diameter in [mm]
electrode.thickness = 2#[2,1,2]  # 1 mm thickness


electrode = tdcs.add_electrode()
electrode.channelnr = 3
electrode.centre = [100,100,0]
electrode.shape = 'ellipse'
electrode.dimensions = [10,10]  # diameter in [mm]
electrode.thickness = 2


tdcs = S.add_tdcslist(deepcopy(tdcs))
tdcs.currents = [0.01, -0.01,0]
tdcs.electrode[0].centre = [100+math.sqrt(9375),75,100]
tdcs.electrode[1].centre = [100+math.sqrt(9375),125,100]
tdcs.electrode[2].centre = [100,100,0]

run_simnibs(S)
<img width="612" alt="微信图片_20241119162705" src="https://github.com/user-attachments/assets/bc227410-fa57-4a00-ab57-43985c8356ae">

**Reply by TorgeW** [reply]
https://github.com/simnibs/simnibs/discussions/478?converting=1
